setup.py

```diff
@@ -256,11 +227,10 @@ def extras_requires(conda_format=False):
     return extras
 
 
-def setup_requirements(requirements_path, module_names, strict_bounds,
```

Comment:
Do we still need the strict_bounds machinery for the conda build case? It seems like that's what this was used for originally. I don't have a good sense of how these changes interact with our existing conda machinery.

Reply:
I don't believe we need it anymore. Let's look at the different cases:

1. -- `install_requires` when invoked by pip --
Expected output is to give wide bounds (i.e. `>=`). We did that by rewriting requirements.txt's `==`, so we never passed `strict_bounds` at the call site, using the default value of `False`.
This is now accomplished by reading requirements.in, which itself uses wide bounds.
A benefit now is that we don't include our transitive dependencies, which makes the install more flexible.
1. -- `install_requires` when invoked by conda-build --
Same as pip case above.
1. -- `setup_requires` when invoked by pip --
`strict_bounds` was `False` in this case, so you'd get latest numpy/cython, which is the whole issue with needing to install the expected `numpy` version in a separate, earlier step. I don't believe it actually matters for cython, but correct me if I'm wrong. This behavior hasn't changed, now that we read from requirements_build.in without rewriting.
1. -- `build_requires` when invoked by conda-build --
`strict_bounds` was `True` in this case, but so was `conda_format`, which overrides any numpy version with `x.x`, so that it's pinned to the version that we specified to conda-build. This way, it will be build with the same version as it requires for install, and the package is tagged with the version. Again, I don't think the behavior for cython matters, but this is a difference here, in that it used to be strict and is no longer. No longer having both `strict_bounds` and `conda_format` as ways to rewrite our requirements is a vast improvement.
1. -- `extras_require` when invoked by pip --
Previously this used `strict_bounds=True`, since the extras were mostly dev requirements, and the thought was to pin them exactly for consistent dev environments. Now this reads from requirements_dev.in, i.e. just lower bounds, and if you want to match the CI environment, you can install using the constraints.
1. -- `extras_require` when invoked by conda-build --
Our conda recipe for zipline doesn't use the `extras_require`.

Reply:
I think those all make sense to me. Thanks for the rundown.